packages/aws-cdk-lib/aws-lambda/lib/params-and-secrets-layers.ts

```diff
@@ -0,0 +1,269 @@
+import { Construct, IConstruct } from 'constructs';
+import { ISecret } from '../../aws-secretsmanager';
+import { IParameter } from '../../aws-ssm';
+import { Token, Stack, Duration } from '../../core';
+import { RegionInfo, FactName } from '../../region-info';
+import * as lambda from '../lib';
+
+/**
+ * Config returned from `ParamsAndSecretsVersion._bind`
+ */
+interface ParamsAndSecretsBindConfig {
+  /**
+   * ARN of the Parameters and Secrets layer version
+   */
+  readonly arn: string;
+
+  /**
+   * Environment variables for the Parameters and Secrets layer configuration
+   */
+  readonly environmentVars: { [key: string]: string };
+}
+
+export enum ParamsAndSecretsVersions {
+  /**
+   * V4
+   *
+   * Note: This is the latest version
+   */
+  V4 = '4',
+}
+
+/**
+ * Logging levels for the Parametes and Secrets Extension
+ */
+export enum ParamsAndSecretsLogLevel {
+  /**
+   * Debug
+   */
+  DEBUG = 'debug',
+
+  /**
+   * Info
+   */
+  INFO = 'info',
+
+  /**
+   * Warn
+   */
+  WARN = 'warn',
+
+  /**
+   * Error
+   */
+  ERROR = 'error',
+
+  /**
+   * No logging
+   */
+  NONE = 'none',
+}
+
+/**
+ * Parameters and Secrets Extension configuration options
+ */
+export interface ParamsAndSecretsOptions {
+  /**
+   * Whether the Parameters and Secrets Extension will cache parameters and
+   * secrets.
+   *
+   * @default true
+   */
+  readonly cacheEnabled?: boolean;
+
+  /**
+   * The maximum number of secrets and parameters to cache. Must be a value
+   * from 0 to 1000. A value of 0 means there is no caching.
+   *
+   * Note: This variable is ignored if parameterStoreTtl and secretsManagerTtl
+   * are 0.
+   *
+   * @default 1000
+   */
+  readonly cacheSize?: number;
+
+  /**
+   * The port for the local HTTP server. Valid port numbers are 1 - 65535.
+   *
+   * @default 2773
+   */
+  readonly httpPort?: number;
+
+  /**
+   * The level of logging provided by the Parameters and Secrets Extension.
+   *
+   * Note: Set to debug to see the cache configuration.
+   *
+   * @default - Logging level will be `info`
+   */
+  readonly logLevel?: ParamsAndSecretsLogLevel;
+
+  /**
+   * The maximum number of connection for HTTP clients that the Parameters and
+   * Secrets Extension uses to make requests to Parameter Store or Secrets
+   * Manager. There is no maximum limit. Minimum is 1.
+   *
+   * Note: This is a per-client configuration.
```

Comment:
I'm not sure people will understand what this means.

Maybe say something like "Every running copy of this Lambda Function may open this many connections, so the total number of connections may exceed this number".

Reply:
Good suggestion. I'll make this change.